iac-extensions/terraform/src/test/java/org/sonar/iac/terraform/parser/BlockTest.java

```diff
@@ -49,8 +60,32 @@ void test_oneLineBlock() {
       .matches("a \"label1\" \"label2\" {}")
       .matches("a \"label with \\\" quote\" {}")
       .matches("a \"label1\" label2 {}")
+      .matches("dynamic a {}")
+      .matches("dynamic a \"label1\" label2 {}")
       .matches("a {b = false}")
       .notMatches("a")
       .notMatches("a{");
   }
+
+  @Test
+  void testDynamicBlock() {
+    var code = """
+      dynamic "a" {
+        for_each = var.a
+        content {
+          b = a.value.b
+        }
+      }
+      """;
+
+    Assertions.assertThat(HclLexicalGrammar.BLOCK)
+      .extracting(parser -> parser.parse(code))
+      .isNotNull()
+      .satisfies(tree -> {
+        var blockTree = (BlockTree) tree;
+        org.assertj.core.api.Assertions.assertThat(blockTree.isDynamic()).isTrue();
```

Comment:
I'm not sure if I saw it, but did you assert that `isDynamic` return `false` in another test?